### PR TITLE
en, zh: Update the compatible TiDB versions for  TiDB Operator

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added, or deleted? (Required)

Update the compatible TiDB versions for  TiDB Operator because TiDB Operator should support all new TiDB releases

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by dveeden — APPROVED** [review]

File: en/_index.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ The corresponding relationship between TiDB Operator and TiDB versions is as fol
 | TiDB Operator version | Compatible TiDB versions |
 |:---|:---|
 | v1.0 | v2.1, v3.0 |
-| v1.1 | v3.0, v3.1, v4.0, v5.0 |
-| dev | v3.0, v3.1, v4.0, v5.0, dev |
+| v1.1 | v3.0 and later releases |
+| dev | v3.0 and later releases, dev |
```
Comment (qiancai) [inline en/_index.md:17]:
```suggestion
| dev | v3.0 and later releases, dev |
```

File: zh/_index.md [hunk 2]
```diff
@@ -15,8 +15,8 @@ TiDB Operator 与适用的 TiDB 版本的对应关系如下：
 | TiDB Operator 版本 | 适用的 TiDB 版本 |
 |:---|:---|
 | v1.0 | v2.1, v3.0 |
-| v1.1 | v3.0, v3.1, v4.0, v5.0 |
-| dev | v3.0, v3.1, v4.0, v5.0, dev |
+| v1.1 | v3.0 及以上版本 |
+| dev | v3.0 及以上版本，dev |
```
Comment (qiancai) [inline zh/_index.md:19]:
```suggestion
| dev | v3.0 及以上版本，dev |
```

**Review by qiancai — COMMENTED** [review]
removed controlled characters

**Pull request merged**